pgql-lang/src/main/java/oracle/pgql/lang/PgqlConfigurationReaderWriter.java

```diff
@@ -0,0 +1,36 @@
+package oracle.pgql.lang;
+
+import java.io.IOException;
+import java.io.Reader;
+
+import org.apache.commons.configuration2.HierarchicalConfiguration;
+import org.apache.commons.configuration2.tree.ImmutableNode;
+import org.apache.commons.vfs2.FileObject;
+import org.metaborg.core.config.AConfigurationReaderWriter;
+import org.metaborg.core.language.LanguageIdentifier;
+import org.metaborg.core.language.LanguageVersion;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.google.inject.ConfigurationException;
+import com.virtlink.commons.configuration2.jackson.JacksonConfiguration;
+
+public class PgqlConfigurationReaderWriter extends AConfigurationReaderWriter {
+
+  @Override
+  protected JacksonConfiguration createNew(HierarchicalConfiguration<ImmutableNode> sourceConfiguration) {
+    return new JacksonConfiguration(new JsonFactory()) {};
+  }
+
+  public JacksonConfiguration create(HierarchicalConfiguration<ImmutableNode> source, FileObject rootFolder) {
+    return createNew(source);
+  }
+
+  @Override
+  public HierarchicalConfiguration<ImmutableNode> read(Reader reader, FileObject rootFolder)
+      throws IOException, ConfigurationException {
+      JacksonConfiguration config = create(null, rootFolder);
+      config.setProperty("name", "pgql-lang");
+      config.setProperty("id", new LanguageIdentifier("oracle.pg", "pgqllang", new LanguageVersion(0,0,0,"SNAPSHOT")));
```

Comment:
Shouldn't it be pgql-lang?

Reply:
When I created the spoofax project back then it did not allow for IDs with a dash in them.

I tried renaming it myself (had to update it a couple places since it's not just this file), but it looks like spoofax uses the ID to generate a bunch of file names and I think Java files cannot have dashes in them. Looks like it's turning the dashes into underscores in some places but not in others since I'm getting the following compile error:

```
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /Users/ovanres/repos/pgql-lang/pgql-spoofax/src-gen/java/pgql_lang/trans/add_type_0_3.java:[201,6] has_timezone_0_0 cannot be resolved
[ERROR] /Users/ovanres/repos/pgql-lang/pgql-spoofax/src-gen/java/pgql_lang/trans/add_type_0_3.java:[223,6] has_timezone_0_0 cannot be resolved
[ERROR] /Users/ovanres/repos/pgql-lang/pgql-spoofax/src-gen/java/pgql_lang/trans/nabl_constraint_0_1_fragment_4.java:[195,6] is_invalid_datetime_0_0 cannot be resolved
[ERROR] /Users/ovanres/repos/pgql-lang/pgql-spoofax/src-gen/java/pgql_lang/trans/nabl_constraint_0_1_fragment_4.java:[355,6] is_invalid_datetime_field_0_0 cannot be resolved
[ERROR] /Users/ovanres/repos/pgql-lang/pgql-spoofax/src-gen/java/pgql_lang/trans/norm_legacy_identifier_0_0_lifted1.java:[12,6] unescape_legacy_identifier_0_0 cannot be resolved
[ERROR] /Users/ovanres/repos/pgql-lang/pgql-spoofax/src-gen/java/pgql_lang/trans/norm_legacy_identifier_0_0_lifted3.java:[12,6] unescape_legacy_string_literal_0_0 cannot be resolved
[ERROR] /Users/ovanres/repos/pgql-lang/pgql-spoofax/src-gen/java/pgql_lang/trans/norm_legacy_string_literal_0_0_lifted1.java:[12,6] unescape_legacy_string_literal_0_0 cannot be resolved
[ERROR] /Users/ovanres/repos/pgql-lang/pgql-spoofax/src-gen/java/pgql_lang/trans/var$Ref_to_legacy_string_literal_0_0_lifted1.java:[12,6] unescape_legacy_identifier_0_0 cannot be resolved
[INFO] 8 errors 
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
```